src/Compilers/CSharp/Portable/Binder/Binder_Crefs.cs

```diff
@@ -216,6 +220,137 @@ private ImmutableArray<Symbol> BindIndexerMemberCref(IndexerMemberCrefSyntax syn
+                parameters = nameMember.Parameters;
+
+                TypeArgumentListSyntax? extensionTypeArguments = syntax.TypeArgumentList;
+                int extensionArity = extensionTypeArguments?.Arguments.Count ?? 0;
+                sortedSymbols = computeSortedAndFilteredCrefExtensionMembers(namedContainer, simpleName.Identifier.ValueText, extensionArity, arity, extensionTypeArguments, diagnostics, syntax);
+            }
+
+            if (sortedSymbols.IsDefaultOrEmpty)
+            {
+                ambiguityWinner = null;
+                return [];
+            }
+
+            Debug.Assert(sortedSymbols.All(s => s.GetIsNewExtensionMember()));
+
+            return ProcessCrefMemberLookupResults(sortedSymbols, arity, syntax, typeArgumentListSyntax, parameters, out ambiguityWinner, diagnostics);
+
+            ImmutableArray<Symbol> computeSortedAndFilteredCrefExtensionMembers(NamedTypeSymbol container, string name, int extensionArity, int arity, TypeArgumentListSyntax? extensionTypeArguments, BindingDiagnosticBag diagnostics, ExtensionMemberCrefSyntax syntax)
+            {
+                Debug.Assert(name is not null);
+
+                LookupOptions options = LookupOptions.AllMethodsOnArityZero | LookupOptions.MustNotBeParameter;
+                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = this.GetNewCompoundUseSiteInfo(diagnostics);
+                ArrayBuilder<Symbol>? sortedSymbolsBuilder = null;
+
+                foreach (var nested in container.GetTypeMembers())
+                {
+                    if (!nested.IsExtension || nested.Arity != extensionArity || nested.ExtensionParameter is null)
+                    {
+                        continue;
+                    }
+
+                    var constructedNested = (NamedTypeSymbol)ConstructWithCrefTypeParameters(extensionArity, extensionTypeArguments, nested);
+
+                    var candidateExtensionSignature = new SignatureOnlyMethodSymbol(
+                         methodKind: MethodKind.Ordinary,
+                         typeParameters: IndexedTypeParameterSymbol.TakeSymbols(constructedNested.Arity),
+                         parameters: [constructedNested.ExtensionParameter],
+                         callingConvention: Cci.CallingConvention.Default,
+                         // These are ignored by this specific MemberSignatureComparer.
+                         containingType: null,
+                         name: null,
+                         refKind: RefKind.None,
+                         isInitOnly: false,
+                         isStatic: false,
+                         returnType: default,
+                         refCustomModifiers: [],
+                         explicitInterfaceImplementations: []);
+
+                    ImmutableArray<ParameterSymbol> extensionParameterSymbols = syntax.Parameters is { } extensionParameterListSyntax ? BindCrefParameters(extensionParameterListSyntax, diagnostics) : default;
```

Comment:
>BindCrefParameters(extensionParameterListSyntax, diagnostics) [](http://example.com/codeflow?start=136&length=61)

It looks like we should attempt to bind parameters even when the enclosing loop never reaches this line.